File: pascal.py
def solution(N):
    result = []

    for i in range(0, N+1):
        result.append([])

        for j in range(0, i + 1):

            if j == 0:
                result[i].append(1)
            elif i == j:
                result[i].append(1)
            else:
                result[i].append(result[i-1][j-1] + result[i-1][j])

    return result[N]

print(solution(3))

File: pascal.cpp
#include <iostream>
#include <vector>
using namespace std;

int N; 
unsigned long long result[41][41];

int main()
{
    cin >> N;

    for (int i = 0; i <= N; i++)
    {
        for (int j = 0; j <= i; j++)
        {
            if (j == 0)
                result[i][j] = 1;
            else if (i == j)
                result[i][j] = 1; 
            else
                result[i][j] = (result[i-1][j-1]) + (result[i-1][j]);
        }
    }

    for (int i = 0; i <= N; i++)
    {
        cout << result[N][i] << endl;
    }

    return 0; 
}